app/models/concerns/mentionable.rb

```diff
@@ -48,20 +48,27 @@ def parse_id_from_decorated_ids(ids)
   end
 
   included do
-    def self.mentionable_users(opts = {})
-      users = active.exclude_user(User.current_user.id)
-                    .with_roles(mentionable_roles(opts[:roles])).distinct.to_a
-      users << add_order_creator(users, opts[:order_id]) if opts[:order_id]
+    def self.mentionable_users(roles:, order_id:, is_private: false)
+      users = User.active.exclude_user(User.current_user.id)
+                  .with_roles(mentionable_roles(roles)).distinct.to_a
+      users << add_order_creator(users, order_id) unless bool_cast.call(is_private)
       users.compact
     end
 
     def self.add_order_creator(users, order_id)
+      valid_roles = User.current_user.roles.pluck(:name) & ['Order administrator', 'Order fulfilment']
```

Comment:
`add_order_creator` is very specific to `order`. Can you use `mentionable_roles` or a different approach?